### PR TITLE
Refactor tests of `F.split_axis` and `F.concat`

### PR DESCRIPTION
**Pull request by toslunar**
Use `chainer.gradient_check`.

### CODE REVIEW

**Comment by hvy** [comment]
jenkins, test this please.

File: tests/chainer_tests/functions_tests/array_tests/test_concat.py [hunk 1]
```diff
@@ -83,22 +90,23 @@ def check_forward(self, inputs, backend_config):
...
+                self._forward, inputs, grad_outputs,
+                **self.check_backward_options)
 
     def test_backward(self, backend_config):
-        self.check_backward(self.inputs, backend_config)
+        grad_outputs = [
+            numpy.random.uniform(
+                -1, 1, self.y_expected.shape
+            ).astype(self.dtype)
+        ]
```
Comment (hvy) [inline tests/chainer_tests/functions_tests/array_tests/test_concat.py]: How about setting `grad_outputs` as a member attribute in `setUp` as you did below. Not a strong opinion but for consistency?

**Comment by hvy** [comment]
jenkins, test this please.

File: tests/chainer_tests/functions_tests/array_tests/test_concat.py (hunk 1 above)
Comment (hvy) [reply tests/chainer_tests/functions_tests/array_tests/test_concat.py]: Thanks for the quick fix!

**Comment by hvy** [comment]
Sorry could you fix the conflicts?

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
LGTM!

**Pull request merged**